### PR TITLE
BUILD: Add xpmem kmod RPM packaging

### PR DESCRIPTION
**Pull request by tvegas1**
Add rpm kmod for xpmem.

### CODE REVIEW

File: ci/azure-pipelines.yml [hunk 1]
```diff
@@ -134,6 +134,23 @@ stages:
...
 
+          - bash: |
+              set -eEx
+              git clean -xdf
+              ./autogen.sh
+              ./configure --disable-kernel-module
+              make dist-gzip
+              mkdir {BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
+              mv xpmem-*.tar.* ./SOURCES/
+              kernel_ver=$(rpm -qa | grep kernel-devel | cut -d'-' -f3-)
```
Comment (yosefe) [inline ci/azure-pipelines.yml]: 1. what happens if we have multiple packages? maybe iterate over all of them?
2. WDYT `rpm -qa 'kernel-devel*'  --qf "%{version}-%{release}.%{arch}\n"`

Comment (tvegas1) [reply ci/azure-pipelines.yml]: fixed both

**Review by brminich — DISMISSED** [review]

**Review by yosefe — DISMISSED** [review]

File: ci/azure-pipelines.yml (hunk 1 above)
Comment (tvegas1) [reply ci/azure-pipelines.yml]: for some reason it does not work on our CI, so falling back to other approach (also rpm -qli seems to fail returning error code when .rpm's are not there)

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -143,7 +143,7 @@ stages:
               mkdir {BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
               mv xpmem-*.tar.* ./SOURCES/
               rpm -qa | grep kernel-devel-
-              for kernel_ver in $(rpm -qa 'kernel-devel-*' --qf "%{version}-%{release}.%{arch}\n"); do
+              for kernel_ver in $(rpm -qa kernel-devel | cut -d'-' -f3-); do
```
Comment (yosefe) [inline ci/azure-pipelines.yml]: rpm -qa 'kernel-devel-*'

**Review by yosefe — CHANGES_REQUESTED** [review]

File: ci/azure-pipelines.yml (hunk 2 above)
Comment (tvegas1) [reply ci/azure-pipelines.yml]: actually we cannot, pls see below, but let me try to re-add `--qf ...`
```
vdi$ rpm -qa kernel-devel
kernel-devel-6.6.12-200.fc39.x86_64
kernel-devel-6.10.10-100.fc39.x86_64

vdi$ rpm -qa 'kernel-devel-*'
kernel-devel-matched-6.10.10-100.fc39.x86_64
<pure kernel-devel is missing here>
```

Comment (tvegas1) [reply ci/azure-pipelines.yml]: ok, without the trailing - it seems to work

Comment (tvegas1) [reply ci/azure-pipelines.yml]: pushed with `'kernel-devel*'` but seems better to match only on exact `kernel-devel`

**Review by yosefe — APPROVED** [review]

**Pull request merged**